data/src/main/scala/cats/data/Validated.scala

```diff
@@ -0,0 +1,143 @@
+package cats
+package data
+
+import cats.data.Or.{LeftOr, RightOr}
+import cats.data.Validated.{Valid, Invalid}
+
+sealed abstract class Validated[+E, +A] extends Serializable {
+  def fold[B](fe: E => B, fa: A => B): B = this match {
+    case Invalid(e) => fe(e)
+    case Valid(a) => fa(a)
+  }
+
+  def isValid: Boolean = fold(_ => false, _ => true)
+  def isInvalid: Boolean = fold(_ => true, _ => false)
+
+  /**
+   * Run the side-effecting function on the value if it is Valid
+   */
+  def foreach(f: A => Unit): Unit = fold(_ => (), f)
+
+  /**
+   * Return the Valid value, or the default if Invalid
+   */
+  def getOrElse[B >: A](default: => B): B = fold(_ => default, identity)
+
+  /**
+   * Is the value Valid and matching the given predicate
+   */
+  def exists(predicate: A => Boolean): Boolean = fold(_ => false, predicate)
+
+  /**
+   * If the value is Valid but the predicate fails, return an empty
+   * Invalid value, otherwise leaves the value unchanged.  This method
+   * is mostly useful for allowing validated values to be used in a
+   * for comprehension with pattern matching.
+   */
+  def filter[EE >: E](pred: A => Boolean)(implicit M: Monoid[EE]): Validated[EE,A] =
+    fold(Invalid.apply, a => if(pred(a)) this else Invalid(M.empty))
+
+  /**
+   * Converts the value to an Either[E,A]
+   */
+  def toEither = fold(Left.apply, Right.apply)
+
+  /**
+   * Returns Valid values wrapped in Some, and None for Invalid values
+   */
+  def toOption: Option[A] = fold(_ => None, Some.apply)
+
+  /**
+   * Convert this value to a single element List if it is Valid,
+   * otherwise return an empty List
+   */
+  def toList: List[A] = fold(_ => Nil, List(_))
+
+  /**
+   * Convert this value to RightOr if Valid or LeftOr if Invalid
+   */
+  def toOr: Or[E,A] = fold(LeftOr.apply,RightOr.apply)
+
+  /**
+   * Validated is a [[Bifunctor]], this method applies one of the
+   * given functions.
+   */
+  def bimap[EE, AA](fe: E => EE, fa: A => AA): Validated[EE, AA] =
+    fold(fe andThen Invalid.apply,
+         fa andThen Valid.apply)
+
+  /**
+   * From Apply:
+   * if both the function and this value are Valid, apply the function
+   */
+  def apply[EE >: E, B](f: Validated[EE, A => B])(implicit EE: Semigroup[EE]): Validated[EE,B] =
+    (this, f) match {
+      case (Valid(a), Valid(f)) => Valid(f(a))
+      case (Invalid(e1), Invalid(e2)) => Invalid(EE.combine(e1,e2))
+      case (e @ Invalid(_), _) => e
+      case (_, e @ Invalid(_)) => e
+
+    }
+
+
+
+  /**
+   * A monadic bind which applies a function if the value is Valid.
+   */
+  def flatMap[EE >: E, B](f: A => Validated[EE,B]): Validated[EE,B] =
```

Comment:
in case this is `Invalid` you can return it and avoid to recreate a new `Validated`